laa-ccms-java-gradle-plugin/src/main/groovy/uk/gov/laa/ccms/gradle/LaaCcmsJavaGradlePlugin.groovy

```diff
@@ -0,0 +1,209 @@
+                def regex = /^[0-9,.v-]+(-r)?$/
+                return stableKeyword || version ==~ regex
+            }
+
+            rejectVersionIf {
+                !isReleaseVersion(it.candidate.version) && isReleaseVersion(it.currentVersion)
+            }
+
+            gradleReleaseChannel = "current"
+        }
+
+        target.jacocoTestReport {
+            sourceDirectories.from target.files('build/generated/sources/annotationProcessor/java/main')
+
+            dependsOn target.tasks['test']
+        }
+
+        target.jacocoTestCoverageVerification {
+
+            violationRules {
+                rule {
+                    limit {
+                        minimum = 0.80
+                    }
+                }
+            }
+
+            dependsOn target.tasks['test']
+        }
+
+        target.checkstyle {
+            maxWarnings = 0
+            toolVersion = CHECKSTYLE_VERSION
+            sourceSets = [sourceSets.main]
+            showViolations = true
+        }
+
+        /** Package repositories **/
+
+        def gitHubPackagesUsername
+        def gitHubPackagesPassword
+
+        def envUsername = System.getenv("GITHUB_ACTOR")?.trim()
+        def envKey = System.getenv("GITHUB_TOKEN")?.trim()
+
+        def propertyUsername = target.findProperty('project.ext.gitPackageUser')
+        def propertyKey = target.findProperty('project.ext.gitPackageKey')
+
+        if (envUsername && envKey) {
+            gitHubPackagesUsername = envUsername
+            gitHubPackagesPassword = envKey
+        } else if (propertyUsername && propertyKey) {
+            gitHubPackagesUsername = propertyUsername
+            gitHubPackagesPassword = propertyKey
+        }
+        else {
+            target.logger.warn("Unable to find GitHub packages credentials. " +
+                    "Please set 'project.ext.gitPackageUser' / 'project.ext.gitPackageKey' " +
+                    "in your gradle.properties file " +
+                    "or 'GITHUB_ACTOR' / 'GITHUB_TOKEN' environment variables.")
+        }
+
+        target.project.ext."gitHubPackagesUsername" = gitHubPackagesUsername
+        target.project.ext."gitHubPackagesPassword" = gitHubPackagesPassword
+
+        target.repositories {
+            mavenCentral()
+
+            // Latest version of saml requires this
+            maven { url "https://build.shibboleth.net/nexus/content/repositories/releases/" }
+
+            // Configure GitHub Packages repositories
+            def githubRepoConfig = { repoUrl ->
+                maven {
+                    url repoUrl
+                    credentials {
+                        username = target.gitHubPackagesUsername
+                        password = target.gitHubPackagesPassword
+                    }
+                }
+            }
+
+            githubRepoConfig('https://maven.pkg.github.com/ministryofjustice/laa-ccms-data-api')
```

Comment:
Should this be in the app using the library, not specified here?

Reply:
same for the other repos below

Reply:
They could be, my thinking was that as this is for all CCMS apps it could be useful to have all the repositories available. It is only CAAB UI that needs them at the moment, so happy to take them out of here if you prefer?